pkg/pillar/zedcloud/send.go

```diff
@@ -437,32 +475,54 @@ func SendOnIntf(ctx ZedCloudContext, destUrl string, intf string, reqlen int64,
 		}
 
 		switch resp.StatusCode {
-		case http.StatusOK:
-			log.Debugf("SendOnIntf to %s StatusOK\n", reqUrl)
-			return resp, contents, false, nil
-		case http.StatusCreated:
-			log.Debugf("SendOnIntf to %s StatusCreated\n", reqUrl)
-			return resp, contents, false, nil
-		case http.StatusNotModified:
-			log.Debugf("SendOnIntf to %s StatusNotModified\n", reqUrl)
-			return resp, contents, false, nil
+		case http.StatusOK, http.StatusCreated, http.StatusNotModified:
+			log.Debugf("SendOnIntf to %s, response %s\n", reqUrl, resp.Status)
+
+			var contents2 []byte
+			var err error
+			rtf := types.ExtStatusNone
+			if ctx.V2API || isCerts { // /certs may not have set the V2API yet
+				if resplen > 0 && checkMimeProtoType(resp) {
+					contents2, rtf, err = verifyAuthentication(contents, isCerts)
+					if err != nil {
+						var envelopeErr bool
+						if rtf == types.ExtStatusHashSizeError || rtf == types.ExtStatusAlgoFail {
+							// server may not support V2 envelope
+							envelopeErr = true
+						}
+						log.Infof("SendOnIntf verify auth error %v, V2 %v, content len %d, url %s, extraStatus %v\n",
+							err, !envelopeErr, len(contents), reqUrl, rtf) // XXX change to debug later
+						if ctx.FailureFunc != nil {
+							ctx.FailureFunc(intf, reqUrl, 0, 0, true)
+						}
+						return nil, nil, rtf, err
+					}
+					log.Infof("SendOnIntf verify auth ok, len content/content2 %d/%d, url %s\n",
```

Comment:
Move to debug in next sprint.
Basically there should be no log.Info when the system is just sitting there (reporting metrics and checking that its config hasn't changed)

Reply:
ok.